src/lib.rs

```diff
@@ -1,744 +1,345 @@
+impl IRShaderReflection {
+    pub fn new(compiler: &IRCompiler) -> Result<IRShaderReflection, Box<dyn std::error::Error>> {
         unsafe {
-            let funcs = IRReflectionFn {
-                create: lib.get(b"IRShaderReflectionCreate")?,
-                destroy: lib.get(b"IRShaderReflectionDestroy")?,
-                get_entry_point_function_name: lib
-                    .get(b"IRShaderReflectionGetEntryPointFunctionName")?,
-                needs_function_constants: lib.get(b"IRShaderReflectionNeedsFunctionConstants")?,
-                get_function_constant_count: lib
-                    .get(b"IRShaderReflectionGetFunctionConstantCount")?,
-
-                copy_function_constants: lib.get(b"IRShaderReflectionCopyFunctionConstants")?,
-                release_function_constants: lib
-                    .get(b"IRShaderReflectionReleaseFunctionConstants")?,
-                copy_compute_info: lib.get(b"IRShaderReflectionCopyComputeInfo")?,
-                copy_vertex_info: lib.get(b"IRShaderReflectionCopyVertexInfo")?,
-                copy_fragment_info: lib.get(b"IRShaderReflectionCopyFragmentInfo")?,
-                copy_geometry_info: lib.get(b"IRShaderReflectionCopyGeometryInfo")?,
-                copy_hull_info: lib.get(b"IRShaderReflectionCopyHullInfo")?,
-                copy_domain_info: lib.get(b"IRShaderReflectionCopyDomainInfo")?,
-                release_compute_info: lib.get(b"IRShaderReflectionReleaseComputeInfo")?,
-                release_vertex_info: lib.get(b"IRShaderReflectionReleaseVertexInfo")?,
-                release_fragment_info: lib.get(b"IRShaderReflectionReleaseFragmentInfo")?,
-                release_geometry_info: lib.get(b"IRShaderReflectionReleaseGeometryInfo")?,
-                release_hull_info: lib.get(b"IRShaderReflectionReleaseHullInfo")?,
-                release_domain_info: lib.get(b"IRShaderReflectionReleaseDomainInfo")?,
-                get_resource_count: lib.get(b"IRShaderReflectionGetResourceCount")?,
-                get_resource_locations: lib.get(b"IRShaderReflectionGetResourceLocations")?,
-            };
-
-            let me = (funcs.create)();
-            Ok(Self { funcs, me })
+            let me = compiler.funcs.IRShaderReflectionCreate();
+            Ok(Self {
+                me,
+                funcs: compiler.funcs.clone(),
+            })
         }
     }
 
     pub fn get_compute_info(
         &self,
         version: IRReflectionVersion,
     ) -> Result<IRVersionedCSInfo, Box<dyn std::error::Error>> {
-        let mut info = unsafe { MaybeUninit::uninit() };
-        if unsafe { (self.funcs.copy_compute_info)(self.me, version, info.as_mut_ptr()) } {
+        let mut info = MaybeUninit::uninit();
+        if unsafe {
+            self.funcs
+                .IRShaderReflectionCopyComputeInfo(self.me, version, info.as_mut_ptr())
+        } {
             Ok(unsafe { info.assume_init() })
         } else {
             panic!("Test me");
```

Comment:
This function returns `Result` for a reason?

Reply:
I'm not sure if we can check for errors on this one, as the docs don't mention it anywhere, and the example code from Apple never checks either. Can we just return the object instead?

Reply:
Postponed to a future PR.